include/nccl_ofi_param_impl.h

```diff
@@ -29,8 +178,17 @@ class ofi_nccl_param_impl {
 		char *envval = getenv(envname);
 		if (envval != NULL) {
 			source = ParamSource::ENVIRONMENT;
-			std::stringstream str(envval);
-			str >> val;
+			auto val_opt = ofi_nccl_param_string_to_value<T>(envval);
+			if (!val_opt) {
+				// object creation happens before plugin init(),
+				// so NCCL_OFI_WARN will not be initialized at
+				// this point.
```

Comment:
For my personal understanding: in the `set` member function of `ofi_nccl_param_impl`, we log errors with `NCCL_OFI_WARN`. Is the assumption that for a `ofi_nccl_param_impl` instance `set` will only be called after the plugin's `init()` sets `ofi_log_function`?